### PR TITLE
Adding the read/write functions

### PR DESCRIPTION
**Pull request by AFulleringer**
Added the read and write functions - need to be validated with actual prompts.

### CODE REVIEW

**Review by gebelangsn — APPROVED** [review]
LGTM! maybe wait for Kyle to merge his branch, then pull from master and push to this branch before merging... (just to make sure no conflicts)

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]: Put this import up with the other imports so it can be sorted in

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]:
```suggestion
        print("creating a new dict for run:", run_title)
```

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]: We should make it possible to choose the directory for this with a flag. Or to save time we should just change it to somewhere not in cwd, maybe `./kvstore` or something?

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]: The way the code works currently, the first argument is the input text for the command, and the second argument is the output from the previous command. We should probably change the way the tools are called across the board, so they all have three arguments: a `run_title`, a `cmd_input`, and a `prev_output`. Mind adding that change into this PR?

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]: These two returns are redundant, and whichever one you keep doesn't need to be in an else block

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py:72]: run the `black` formatter real quick

File: src/affordance/tasks/tools.py
Comment (kylrth) [inline src/affordance/tasks/tools.py]: Maybe let's move some of these example comments into docstrings or something

**Review by kylrth — CHANGES_REQUESTED** [review]
I've got some suggestions! Isn't code review fun

**Comment by kylrth** [comment]
Oh also don't forget to add this bit:

```python
    "[read]": read,
    "[write]": write,
    "[list keys]": list_keys,
```

In the tool dict

**Comment by AFulleringer** [comment]
Somehow the changes to add run_title to visit and its lineage ended up in 'reduce cost of debug config' - just a heads up on where to look for that.

File: .vscode/launch.json
Comment (kylrth) [inline .vscode/launch.json]: Haha somehow a merge conflict ended up here. It sounds like you have `commit`-ment issues!

I'm going to add one more commit to remove some of this stuff, and then when we're finished we can squash everything down into a single commit for posterity.

File: src/affordance/tasks/arxiv_summarization.py
Comment (kylrth) [inline src/affordance/tasks/arxiv_summarization.py]: I actually think it's better to have a PR contain just one new thing at a time. Better to leave Georges' stuff in his branch

File: src/affordance/tasks/sequential_interpreter.py
Comment (kylrth) [inline src/affordance/tasks/sequential_interpreter.py:47]: Sometimes this function is called with a batch across multiple separate evaluations, so we probably need to make `run_title` be a `List[str]`.

File: src/affordance/tasks/sequential_interpreter.py
Comment (kylrth) [inline src/affordance/tasks/sequential_interpreter.py]: looks like I forgot to mention that I've been using `black --line-length 100`, which allows these lines to be longer. I'll fix that

File: src/affordance/tasks/sequential_interpreter.py
Comment (kylrth) [inline src/affordance/tasks/sequential_interpreter.py:193]: We should log the run_title too

**Comment by kylrth** [comment]
Alright once you make those few last changes, the last thing we need is to squash:

```sh
git fetch
git reset --soft origin/master
git commit -m "<a nice summary>"
git push -f
```

**Review by kylrth — APPROVED** [review]

File: .github/workflows/black.yml
Comment (kylrth) [inline .github/workflows/black.yml]: You'll want to keep this

**Review by kylrth — CHANGES_REQUESTED** [review]

**Review by kylrth — APPROVED** [review]

**Pull request merged**
